packages/grafana-schema/src/veneer/dashboard.types.ts

```diff
@@ -0,0 +1,32 @@
+import * as raw from '../raw/dashboard/x/dashboard.gen';
+
+export interface Dashboard extends raw.Dashboard {
+  panels?: Array<
+    | Panel
+    | raw.RowPanel
+    | {
+        type: 'graph';
+      }
+    | {
+        type: 'heatmap';
+      }
```

Comment:
Basically, those are there because AIUI old angular plugins would spew their config all over the root of the panel object, rather than being confined to a few subtrees. I had to make exceptions for them in order to allow that (it's impossible to do from a normal plugin `models.cue` file, very much by design), and this was the way i found.

But that predates this issue, and once this PR is in and we have a tight dev loop established, we should (finally!) be able to really find a solution we're happy with for that iteratively/during some mobbing.